pkg/storage/sqlstorage/store.go

```diff
@@ -54,4 +63,33 @@ func (s *Store) Close(ctx context.Context) error {
 	return s.onClose(ctx)
 }
 
+func (s *Store) withSqlTx(ctx context.Context, callback func(tx *sql.Tx) error) error {
+	tx, err := s.schema.BeginTx(ctx, &sql.TxOptions{})
+	if err != nil {
+		return s.error(err)
+	}
+	defer func() {
+		_ = tx.Rollback()
+	}()
+	err = callback(tx)
+	if err != nil {
+		return s.error(err)
+	}
+	return s.error(tx.Commit())
+}
+
+func (s *Store) WithTX(ctx context.Context, callback func(api storage.API) error) error {
+	return s.withSqlTx(ctx, func(tx *sql.Tx) error {
+		return callback(NewAPI(s.schema, tx))
+	})
+}
+
+func NewStore(schema Schema, onClose func(ctx context.Context) error) *Store {
+	return &Store{
+		API:     NewAPI(schema, schema),
```

Comment:
This is weird 😅

Reply:
Yes, it works, maybe it will need a bit of rework.
The purpose of this PR is to remove triggers, we have to ensure it is working as expected, not lost too many time on implementation details.

Reply:
Alright :)